### PR TITLE
PETSc: implement array_from_vector and gather methods

### PR DESCRIPTION
**Pull request by dajuno**

### CODE REVIEW

File: transiflow/interface/PETSc.py [hunk 1]
```diff
@@ -55,6 +65,10 @@ def __init__(self, parameters, nx, ny, nz, dim, dof, comm=PETSc.COMM_WORLD):
             set(self.assembly_map.indices).difference(self.map_natural.indices)
         )
 
+        indices_natural_vec = Vector().createWithArray(self.map_natural.indices)
+        sct, self.indices_natural_global_vec = PETSc.Scatter().toAll(indices_natural_vec)
```
Comment (Sbte) [inline transiflow/interface/PETSc.py:69]: Should the name be `index_natural_global_permut` so it matches the other one?

File: transiflow/interface/PETSc.py [hunk 2]
```diff
@@ -77,21 +91,19 @@ def vector(self):
...
-           locations of the ghosts in the sequential vector. The locations correspond to
-           the ghost indices in PETSc ordering.
-        """
-        vec_indices = Vector().createWithArray(self.map_natural.indices)
-        sct, vec_global_indices = PETSc.Scatter().toAll(vec_indices)
-        sct.scatter(vec_indices, vec_global_indices)
+    def array_from_vector(self, vector):
+        array = Vector.all_gather(vector).array
+        array_natural = array.copy()
+        array_natural[self.indices_natural_global_vec.array.astype(PETSc.IntType)] = array[:]
```
Comment (Sbte) [inline transiflow/interface/PETSc.py:97]: I thought there was some numpy function that did this, but I also can't find it.

**Review by Sbte — CHANGES_REQUESTED** [review]

File: transiflow/interface/PETSc.py (hunk 2 above)
Comment (dajuno) [reply transiflow/interface/PETSc.py:97]: do you mean `numpy.argsort`?

Comment (Sbte) [reply transiflow/interface/PETSc.py:97]: That could be the one I was thinking off, yeah.

Comment (dajuno) [reply transiflow/interface/PETSc.py]: yes, that's what I used for the other permutation vector. looks better and more consistent

Comment (Sbte) [reply transiflow/interface/PETSc.py]: That looks great! Can you squash the first and last commit? Then I'll merge it.

Comment (dajuno) [reply transiflow/interface/PETSc.py]: sure, done

**Review by Sbte — APPROVED** [review]

**Pull request merged**